src/components/multi_file_viewer.vue

```diff
@@ -51,23 +52,65 @@
     files_currently_viewing: OpenFile[] = [];
     active_tab_index = 0;
 
-    add_to_viewing(filename: string, file_contents: string) {
+    async add_to_viewing(filename: string, file_contents: string | (() => Promise<string>),
```

Comment:
Note to self (or to Ashley if you want to take a crack at this change): 
- Change file_contents type to `string | Promise<string>`
- Pass the  `string | Promise<string>` to view-file (view-file should be responsible for awaiting on the promise
- Add a loading icon in view-file (loading = true; await file_contents; loading = false)
- Calls to add_to_viewing that currently pass in a function can remove the lambda and just pass the promise